source/tanya/math/random.d

```diff
@@ -160,6 +160,181 @@ version (linux)
         }
     }
 }
+else version (CRuntime_Bionic) version = secure_arc4random;
+else version (OSX) version = secure_arc4random;
+else version (OpenBSD) version = secure_arc4random;
+else version (NetBSD) version = secure_arc4random;
+// Uncomment the following line if not supporting Solaris versions prior to 11.3.
+//else version (Solaris) version = secure_arc4random;
+
+version (secure_arc4random)
+{
+    private extern (C) void arc4random_buf(scope void* buf, size_t nbytes) nothrow @nogc @system;
+
+    /**
+     * Uses arc4random_buf.
+     */
+    class PlatformEntropySource : EntropySource
+    {
+        /**
+         * Returns: Minimum bytes required from the entropy source.
+         */
+        override @property ubyte threshold() const pure nothrow @safe @nogc
+        {
+            return 32;
+        }
+
+        /**
+         * Returns: Whether this entropy source is strong.
+         */
+        override @property bool strong() const pure nothrow @safe @nogc
+        {
+            return true;
+        }
+
+        /**
+         * Poll the entropy source.
+         *
+         * Params:
+         *  output = Buffer to save the generate random sequence (the method will
+         *           to fill the buffer).
+         *
+         * Returns: Number of bytes that were copied to the $(D_PARAM output)
+         *          or $(D_PSYMBOL Nullable!ubyte.init) on error.
+         */
+        override Nullable!ubyte poll(out ubyte[maxGather] output) nothrow @nogc @safe
+        {
+            (() @trusted => arc4random_buf(output.ptr, output.length))();
+            return Nullable!ubyte(cast(ubyte) (output.length));
+        }
+    }
+
+    private unittest
+    {
+        auto entropy = defaultAllocator.make!Entropy();
+        ubyte[blockSize] output;
+        output = entropy.random;
+
+        defaultAllocator.dispose(entropy);
+    }
+}
+else version (Windows)
+{
+    import core.sys.windows.basetsd : ULONG_PTR;
+    import core.sys.windows.windef : BOOL, DWORD, PBYTE;
+    import core.sys.windows.winnt : LPCSTR, LPCWSTR;
+    import core.sys.windows.wincrypt : HCRYPTPROV;
+    private extern(Windows) @nogc nothrow
+    {
+        BOOL CryptGenRandom(HCRYPTPROV, DWORD, PBYTE);
+        BOOL CryptAcquireContextA(HCRYPTPROV*, LPCSTR, LPCSTR, DWORD, DWORD);
+        BOOL CryptAcquireContextW(HCRYPTPROV*, LPCWSTR, LPCWSTR, DWORD, DWORD);
+        BOOL CryptReleaseContext(HCRYPTPROV, ULONG_PTR);
+    }
+
+    private bool initCryptGenRandom(scope ref HCRYPTPROV hProvider) @nogc nothrow @trusted
+    {
+        import core.sys.windows.winbase : GetLastError;
+        import core.sys.windows.winerror : NTE_BAD_KEYSET;
+        import core.sys.windows.wincrypt : PROV_RSA_FULL, CRYPT_NEWKEYSET, CRYPT_VERIFYCONTEXT, CRYPT_SILENT;
```

Comment:
We can spare one line if we remove this last import and import the whole `core.sys.windows.wincrypt` above. It hasn't raised any name conflicts on my win10 machine.